src/hub/oauth_action.ts

```diff
@@ -0,0 +1,13 @@
+import {Action} from "./action"
+import {ActionRequest} from "./action_request"
+
+export abstract class OAuthAction extends Action {
+  usesOauth = true
+  abstract async oauthCheck(request: ActionRequest): Promise<boolean>
+  abstract async oauthUrl(redirectUri: string, encryptedState: string): Promise<string>
+  abstract async oauthFetchInfo(urlParams: { [key: string]: string }, redirectUri: string): Promise<string>
+}
+
+export function isOauthAction(action: Action): boolean {
```

Comment:
I would think this function belongs on the action itself instead of as a separate export or does not violate the statelessness?